pkg/common/common.go

```diff
@@ -56,28 +60,29 @@ func FetchEnv(key string, def string) string {
 	return val
 }
 
+// GetDefaultIfNil returns the value of a pointer argument, or a default value if the pointer is nil.
 func GetDefaultIfNil[T any](val *T, def T) T {
 	if reflect.ValueOf(val).IsNil() {
 		return def
 	}
 	return *val
 }
 
+// GetEmptySliceIfNil returns a provided slice, or an empty slice of the same type if the input slice is nil.
 func GetEmptySliceIfNil[T any](val []T) []T {
-	if val == nil {
```

Comment:
This refactor is not wrong, per se, because it does not change behavior. So my question is why the change. I might be missing something. 

If the input slice is empty, it will enter the `if` branch and it will return a new empty slice. While in the "old" implementation, the method would return the same empty input slice.

Reply:
The change from 'val == nil' to 'len(val) == 0' in the if statement really doesn't affect the behavior of the code. I think, I wanted to change only the return statement, but left the len-check because this was the last thing I was also benchmarking and they were almost identical. So, you are right, thanks for noticing that! Do you want me to revert this change in if-statement?

Reply:
Yes, revert the change. By rule of thumb, never change code while writing tests unless you find a bug ;)